pkg/utils/errors/errors.go

```diff
@@ -1,18 +1,55 @@
 package errors
 
 import (
+	"os"
+
 	log "github.com/sirupsen/logrus"
 )
 
-// CheckError is a convenience function to exit if an error is non-nil and exit if it was
-func CheckError(err error) {
+const (
+	// ErrorCommandSpecific is reserved for command specific indications
+	ErrorCommandSpecific = 1
+	// ErrorConnectionFailure is returned on connection failure to API endpoint
+	ErrorConnectionFailure = 11
+	// ErrorAPIResponse is returned on unexpected API response, i.e. authorization failure
+	ErrorAPIResponse = 12
+	// ErrorResourceDoesNotExist is returned when the requested resource does not exist
+	ErrorResourceDoesNotExist = 13
+	// ErrorGeneric is returned for generic error
+	ErrorGeneric = 20
+)
```

Comment:
We should think about whether these codes are sufficient and if the classifications are correct. The code where I left off was clearly not finished yet and not thought through to the end, so I'm unsure whether we should adapt it in this state in a verbatim way

Reply:
I have used these 5 error classes for argo-cd in https://github.com/argoproj/argo-cd/pull/3700 and it seems okay. I guess we'll truly know if they are sufficient in you can review that PR 😅